modules/menuItems.js

```diff
@@ -224,40 +264,9 @@ let menuTempl = function (webviews) {
                     },
                 ],
             },
-            {
-                type: 'separator',
-            },
-            {
-                label: i18n.t('mist.applicationMenu.file.swarmUpload'),
-                accelerator: 'Shift+CommandOrControl+U',
-                enabled: store.getState().settings.swarmState == SwarmState.Enabled,
-                click() {
-                    const focusedWindow = BrowserWindow.getFocusedWindow();
-                    const paths = dialog.showOpenDialog(focusedWindow, {
-                        properties: ['openFile', 'openDirectory']
-                    });
-                    if (paths && paths.length === 1) {
-                        const isDir = fs.lstatSync(paths[0]).isDirectory();
-                        const defaultPath = path.join(paths[0], 'index.html');
-                        const uploadConfig = {
-                            path: paths[0],
-                            kind: isDir ? 'directory' : 'file',
-                            defaultFile: fs.existsSync(defaultPath) ? '/index.html' : null
-                        };
-                        swarmNode.upload(uploadConfig).then((hash) => {
-                            focusedWindow.webContents.executeJavaScript(`
-                              Tabs.update('browser', {$set: {
-                                  url: 'bzz://${hash}',
-                                  redirect: 'bzz://${hash}'
-                              }});
-                              LocalStore.set('selectedTab', 'browser');
-                            `);
-                            swarmLog.info('Hash uploaded:', hash);
-                        }).catch(e => swarmLog.error(e));
-                    }
-                }
-            }]
-    });
+            ...swarmUpload
```

Comment:
Just to unpack the array of the [seperator, swarmUpload_menu_item] into the submenu where it’s located. And if it’s null (not set because in wallet mode) then it doesn’t add any items.